src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,227 @@
+
+            } catch (DukeException error) {
+                printErrorMessage(PROMPT_TASK_NUMBER);
+
+            } catch (ArrayIndexOutOfBoundsException error) {
+                printErrorMessage(PROMPT_SENSIBLE_INDEX);
+
+            } catch (NumberFormatException error) {
+                printErrorMessage(PROMPT_SENSIBLE_INDEX);
+
+            }
+            break;
+        case "todo":
+            try {
+                executeTaskCase(input, TODO_STARTING_INDEX, TaskType.TODO);
+
+            } catch (StringIndexOutOfBoundsException error) {
+                printErrorMessage(PROMPT_TASK_DESCRIPTION);
+            }
+            break;
+        case "deadline":
+            try {
+                executeTaskCase(input, DEADLINE_STARTING_INDEX,TaskType.DEADLINE);
+
+            } catch (StringIndexOutOfBoundsException error) {
+                printErrorMessage(PROMPT_TASK_DESCRIPTION);
+
+            }
+            break;
+        case "event":
+            try {
+                executeTaskCase(input, EVENT_STARTING_INDEX,TaskType.EVENT);
+
+            } catch (StringIndexOutOfBoundsException error) {
+                printErrorMessage(PROMPT_TASK_DESCRIPTION);
+            }
+            break;
+        default:
+            printErrorMessage(TYPE_SUITABLE_COMMAND_MESSAGE);
+        }
+    }
+
+    private static void printTaskMessage() {
+        printIndentationAndDivider();
+        printWordsWithIndentation(ADDED_TO_LIST);
+        printWordsWithIndentation(tasks[Task.getTotalTasks() - 1].getStatusIconAndDescription());
+        printWordsWithIndentation(notifyNumberOfTasks());
+        printIndentationAndDivider();
+        System.out.println();
+    }
+
+    private static void printErrorMessage(String error_message) {
+        printIndentationAndDivider();
+        printWordsWithIndentation(error_message);
+        printIndentationAndDivider();
+        System.out.println();
+    }
+
+    private static void executeTaskCase(String input, int starting_index, TaskType type) {
+        String description = input.strip().substring(starting_index);
+        try {
+            addTask(description, type);
+            printTaskMessage();
+
+        } catch (DukeException error) {
+            printErrorMessage(MISSING_ARGUMENTS_FOR_EVENT_AND_DEADLINE);
+        }
+    }
+
+    private static String[] parseInputForDifferentTask(String input,TaskType type) {
+        String[] parsedOutput = {};
+        switch(type) {
+        case TODO:
+            parsedOutput = new String[]{input};
+            break;
+        case DEADLINE:
+            parsedOutput = input.split(DEADLINE_DESCRIPTION_AND_DATE_SPLITTER);
+            break;
+        case EVENT:
+            parsedOutput = input.split(EVENT_DESCRIPTION_AND_DATE_TIME_SPLITTER);
+            break;
+        }
+        return parsedOutput;
+    }
+
+    private static void addTask(String input,TaskType type) throws StringIndexOutOfBoundsException, DukeException {
```

Comment:
Suggest to relocate the function definition from "throw" onwards to a new line with appropriate line indent.